lockfile/src/lib.rs

```diff
@@ -22,6 +23,9 @@ mod parsers;
 mod python;
 mod ruby;
 
+/// Maximum directory depth to recurse for finding lockfiles.
+const MAX_LOCKFILE_DEPTH: usize = 5;
```

Comment:
I've picked 5 here because I like the number 5. a depth of 10 seems like it would already be kinda deep for a lockfile and 3 seemed somewhat shallow. It probably isn't too big of a deal to go even deeper, I don't think this should be too much of a performance problem, but I'm not sure it's necessary.